packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts

```diff
@@ -362,4 +362,13 @@ function readIfPossible(filename: string): string | undefined {
     debug(e);
     return undefined;
   }
+}
+
+/**
+ * Return the username with characters invalid for a RoleSessionName removed
+ *
+ * @see https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html#API_AssumeRole_RequestParameters
+ */
+function safeUsername() {
+  return os.userInfo().username.replace(/[^\w+=,.@-]/g, '@');
 }
```

Comment:
Move this into a reusable spot? `os.ts` sounds appropriate?

Reply:
If you are addressing the repetition in this PR, please note that the changes are in 2 different packages and there's no good shared package to put this functionality in otherwise.

If you are addressing potential future reuse, I would argue against optimizing for that beforehand.